benches/benches.rs

```diff
@@ -30,36 +31,75 @@ async fn ws_server(tx: Sender<SocketAddr>) {
 	}
 }
 
-pub fn http(c: &mut criterion::Criterion) {
-	c.bench_function("http 100 requests", |b| {
-		let (tx_addr, rx_addr) = oneshot::channel::<SocketAddr>();
-		async_std::task::spawn(http_server(tx_addr));
-		let server_addr = block_on(rx_addr).unwrap();
-		let client = HttpClient::new(&format!("http://{}", server_addr));
+pub fn http_requests(c: &mut criterion::Criterion) {
+	let mut rt = tokio::runtime::Runtime::new().unwrap();
+	let (tx_addr, rx_addr) = oneshot::channel::<SocketAddr>();
+	async_std::task::spawn(http_server(tx_addr));
+	let server_addr = block_on(rx_addr).unwrap();
+	let client = Arc::new(HttpClient::new(&format!("http://{}", server_addr)));
 
+	c.bench_function("synchronous http round trip", |b| {
 		b.iter(|| {
-			block_on(async {
-				for _ in 0..100 {
-					let _: JsonValue = black_box(client.request("say_hello", Params::None).await.unwrap());
-				}
+			rt.block_on(async {
+				let _: JsonValue = black_box(client.request("say_hello", Params::None).await.unwrap());
 			})
 		})
 	});
+
+	c.bench_function_over_inputs(
+		"concurrent http round trip",
+		move |b: &mut Bencher, size: &usize| {
+			b.iter(|| {
+				let mut tasks = Vec::new();
+				for _ in 0..*size {
+					let client_rc = client.clone();
+					let task = rt.spawn(async move {
+						let _: Result<JsonValue, _> = black_box(client_rc.request("say_hello", Params::None)).await;
+					});
+					tasks.push(task);
+				}
```

Comment:
Can you move this part outside the `b.iter()` block to measure only the task execution?

Reply:
No, we spawn `n concurrent tasks` that we need to wait for to complete.

The actual joining of `tasks` are not great because a task spawned "later" could potentially be finished before....

Reply:
This compiles, but panics as soon as it runs:

```rust
			let mut tasks = Vec::new();
			for _ in 0..*size {
				let client_rc = client.clone();
				let task = rt.spawn(async move {
					let _: Result<JsonValue, _> = black_box(client_rc.request("say_hello", Params::None)).await;
				});
				tasks.push(task);
			}
			b.iter(|| {
				for mut task in &mut tasks {
					rt.block_on(&mut task).unwrap();
				}
			})
```

…so I guess you're right! :)

Reply:
You mean the `Vec` correct? I think so but then each iteration has to clear/replace it with an empty `Vec`.

Reply:
That should work AFAIU.